pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
+	// Required: The IP of this endpoint.
+	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
```

Comment:
Did we permit hostname before? For Endpoint, in particular, I think it's best if it's an address, to avoid dependencies on DNS.

We should at least allow v6 as well as v4.